grakn-engine/src/main/java/ai/grakn/engine/controller/ImportController.java

```diff
@@ -211,47 +207,48 @@ private void importDataFromFile(String dataFile, Loader loaderParam) {
         }
     }
 
-    private void consumeEntity(Var var, Loader loader) {
-        if (!entitiesMap.containsKey(var.admin().getName()) && isIndependentEntity(var)) {
-            if (var.admin().isUserDefinedName()) {
-                // Some variable might not have an explicit ID defined, in that case we generate explicitly one and we save it into our cache
-                // so that we can refer to it.
-                String varId = (var.admin().getId().isPresent()) ? var.admin().getId().get() : UUID.randomUUID().toString();
-                entitiesMap.put(var.admin().getName(), varId);
-                // We force the ID of the current var to be the one computed by this controller.
-                loader.add(Graql.insert(var.admin().id(varId)));
-            } else {
-                loader.add(Graql.insert(var));
-            }
-            processedEntities.incrementAndGet();
+    private Object consumeInsertEntity(Iterator<Object> batchIterator, Loader loader) {
+        Object var = null;
+        List<Var> insertQuery = new ArrayList<>();
+        while (batchIterator.hasNext()) {
+            var = batchIterator.next();
+            if (var instanceof Var) {
+                insertQuery.add(((Var) var));
+                processedEntities.incrementAndGet();
+            } else
+                break;
```

Comment:
Why is there a break here ? What does `batchIterator` contain that is not a `var` ?